### PR TITLE
feat(share/eds)!: Don't use json.Marshal for inverted index

### PR DESCRIPTION
**Pull request by walldiss**
## Overview

Inverted index stores shard keys as json objects. `json.Marshal` and `json.Unmarshal` are memory heavy operations and slows down inverted index building. Shard keys are strings and could be stored and retrieved as is, without marshalling/unmarshallng. 

Benchmarks shows 15% performance gain for `store.Put` operation.

### CODE REVIEW

File: share/eds/store.go [hunk 1]
```diff
@@ -200,6 +209,7 @@ func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.Ext
 		return ctx.Err()
 	case result := <-ch:
 		if result.Error != nil {
+			fmt.Println("ERROR", result.Error)
```
Comment (distractedm1nd) [inline share/eds/store.go:212]: snuck in

File: share/eds/store.go [hunk 2]
```diff
@@ -182,15 +182,24 @@ func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.Ext
...
 	if err != nil {
 		return fmt.Errorf("failed to write EDS to file: %w", err)
 	}
 
+	// write whole buffer in one go to optimise i/o
+	if _, err = f.Write(buf.Bytes()); err != nil {
+		return fmt.Errorf("failed to write EDS to file: %w", err)
+	}
+
+	mount := &inMemoryOnceMount{
```
Comment (distractedm1nd) [inline share/eds/store.go:197]: The memory mount stuff is a different or no?

**Review by distractedm1nd — DISMISSED** [review]

**Review by renaynay — COMMENTED** [review]
is this breaking?

**Comment by renaynay** [comment]
@walldiss can you base this on feature branch on main repo and we can merge it in there?

**Review by vgonkivs — DISMISSED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — DISMISSED** [review]
Please do not merge yet 🙏🏻

**Review by distractedm1nd — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**